internal/controller/atlasdatafederation/datafederation_controller_test.go

```diff
@@ -237,7 +237,11 @@ func TestDeleteConnectionSecrets(t *testing.T) {
 				gotSecrets.Items[i].ResourceVersion = ""
 			}
 			assert.NoError(t, err)
-			assert.Equal(t, tc.wantSecrets, gotSecrets.Items)
+			wantSecrets := tc.wantSecrets
+			if wantSecrets == nil {
```

Comment:
Test case should set `nil` instead of have logic around it.

Reply:
Tried to do that, but the linter was complaining about duplicated code between test cases... I can give it another go.

Reply:
Hey, turns out the test case was really duplicated!gt